crates/ra_hir/src/ty.rs

```diff
@@ -20,11 +20,8 @@ pub(crate) use lower::{TypableDef, CallableDef, type_for_def, type_for_field, ca
 pub(crate) use infer::{infer, InferenceResult, InferTy};
 use display::{HirDisplay, HirFormatter};
 
-/// A type. This is based on the `TyKind` enum in rustc (librustc/ty/sty.rs).
-///
-/// This should be cheap to clone.
-#[derive(Clone, PartialEq, Eq, Debug)]
-pub enum Ty {
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub enum TypeName {
```

Comment:
Isn't a `TypeCtor` a better name here? I imagine name would be a pretty overloaded term, so it's better to avoid it. I guess `TypeName` is what's used in chalk? I wonder if it was intentionally chosen over TypeCtor.

Reply:
Renamed it to `TypeCtor` now.